presto-main/src/main/java/com/facebook/presto/sql/planner/SystemPartitioningHandle.java

```diff
@@ -44,24 +44,108 @@
 public final class SystemPartitioningHandle
         implements ConnectorPartitioningHandle
 {
+    public static Partitioning singlePartition()
+    {
+        return createSystemPartitioning(SystemPartitioning.FIXED, SystemPartitionFunction.SINGLE);
+    }
+
+    public static boolean isSystemSinglePartitioning(Partitioning partitioning)
+    {
+        return isSystemPartitioning(partitioning, SystemPartitionFunction.SINGLE);
+    }
+
+    public static boolean isSystemSinglePartitioning(PartitioningHandle partitioningHandle)
+    {
+        return isSystemPartitioning(partitioningHandle.getConnectorHandle(), SystemPartitionFunction.SINGLE);
+    }
+
+    public static Partitioning coordinatorOnlyPartition()
+    {
+        return createSystemPartitioning(SystemPartitioning.COORDINATOR_ONLY, SystemPartitionFunction.SINGLE);
+    }
+
+    public static Partitioning fixedHashPartitioning(List<Symbol> columns)
+    {
+        return createSystemPartitioning(SystemPartitioning.FIXED, SystemPartitionFunction.HASH, columns);
+    }
+
+    public static boolean isFixedHashPartitioning(Partitioning partitioning)
+    {
+        return isSystemPartitioning(partitioning, SystemPartitionFunction.HASH);
+    }
+
+    public static Partitioning fixedRandomPartitioning()
+    {
+        return createSystemPartitioning(SystemPartitioning.FIXED, SystemPartitionFunction.ROUND_ROBIN);
+    }
+
+    public static boolean isFixedRandomPartitioning(Partitioning partitioning)
+    {
+        return isSystemPartitioning(partitioning, SystemPartitionFunction.ROUND_ROBIN);
+    }
+
+    public static Partitioning fixedBroadcastPartitioning()
+    {
+        return createSystemPartitioning(SystemPartitioning.FIXED, SystemPartitionFunction.BROADCAST);
+    }
+
+    public static boolean isFixedBroadcastPartitioning(Partitioning partitioning)
+    {
+        return isSystemPartitioning(partitioning, SystemPartitionFunction.BROADCAST);
+    }
+
+    public static Partitioning unknownStreamPartitioning()
```

Comment:
It is, but that is an internal detail.  The problem is the only way to say "I have no defined node partitioning is to say SOURCE".  Then there are a bunch of places that depend on this.  This is something we should clean up in the future, which is why I added an easy to read method name to distinguish the two cases.